Sources/Sentry/SentrySessionReplayIntegration.m

```diff
@@ -114,39 +115,40 @@ - (void)resumePreviousSessionReplay:(SentryEvent *)event
         }
     }
 
-    _resumeReplayMaker = [[SentryOnDemandReplay alloc] initWithContentFrom:lastReplayURL.path];
-    _resumeReplayMaker.bitRate = _replayOptions.replayBitRate;
-    _resumeReplayMaker.videoScale = _replayOptions.sizeScale;
+    SentryOnDemandReplay *resumeReplayMaker =
+        [[SentryOnDemandReplay alloc] initWithContentFrom:lastReplayURL.path];
+    resumeReplayMaker.bitRate = _replayOptions.replayBitRate;
+    resumeReplayMaker.videoScale = _replayOptions.sizeScale;
 
     NSDate *beginning = hasCrashInfo
         ? [NSDate dateWithTimeIntervalSinceReferenceDate:crashInfo.lastSegmentEnd]
-        : [_resumeReplayMaker oldestFrameDate];
+        : [resumeReplayMaker oldestFrameDate];
 
     if (beginning == nil) {
         return; // no frames to send
     }
 
-    NSError *error;
-    if (![_resumeReplayMaker
+    _dispatchQueue = [[SentryDispatchQueueWrapper alloc] init];
+
+    [_dispatchQueue dispatchAsyncWithBlock:^{
+        SentryReplayType _type = type;
+        int _segmentId = segmentId;
+
+        NSError *error;
+        NSArray<SentryVideoInfo *> *videos = [resumeReplayMaker
             createVideoWithBeginning:beginning
                                  end:[beginning dateByAddingTimeInterval:duration]
-                       outputFileURL:[lastReplayURL URLByAppendingPathComponent:@"lastVideo.mp4"]
-                               error:&error
-                          completion:^(SentryVideoInfo *video, NSError *renderError) {
-                              if (renderError != nil) {
-                                  SENTRY_LOG_ERROR(
-                                      @"Could not create replay video: %@", renderError);
-                              } else {
-                                  [self captureVideo:video
-                                            replayId:replayId
-                                           segmentId:segmentId
-                                                type:type];
-                              }
-                              self->_resumeReplayMaker = nil;
-                          }]) {
-        SENTRY_LOG_ERROR(@"Could not create replay video: %@", error);
-        return;
-    }
+                               error:&error];
+        if (videos == nil) {
```

Comment:
Good point.

Two cases here:

- The previous replay was in session mode: The replay exists, and we have to give the event the replay_id, regardless of whether the final segment works or not.
- The previous replay was in buffer mode: We need to compile the video and set the replay_id of the event after having the video. It will show an error message on the issue page that the replay is not available if we set the replay_id and the video is nil.

I chose the second option because I didn't want to hold the EventProcessor until the video is rendered. If it's okay to hold the EventProcessor, we can change this, but in a PR focused on it, this behavior didn't change in this PR.

Reply:
> It will show an error message on the issue page that the replay is not available if we set the replay_id and the video is nil.

That will be confusing to our users.

If I got it correctly, `resumePreviousSessionReplay` only runs for crash events. The SDK captures crash events on a background thread, unless it's a start-up crash see https://github.com/getsentry/sentry-cocoa/blob/649d9405632d115712a1e4135287f311df713688/Sources/Sentry/SentryCrashReportSink.m#L48-L64

So `resumePreviousSessionReplay` usually already runs on a background thread, meaning you could remove the dispatch queue to avoid this problem. You only have to point that out clearly with a comment.

Reply:
Ok. Lets do this.